### DISCUSSION TITLE
CP2K Release 2022.1

### DISCUSSION THREAD

**Discussion by oschuett** [cp2k/cp2k#2611] [Team Posts]
Hi Everybody,

as discussed at our last developers meeting, we'd like to officially adopt a 6 month release cycle and switch to [Calendar Versioning](https://calver.org). For our next release we're targeting the end of June.

So, please merge your changes and update the [version history](https://www.cp2k.org/version_history) in time.

At the moment I'm not aware of any blocking issues.

**Comment by alazzaro** [comment]
@oschuett DBCSR release will be released in 2-3 weeks, so I'm asking if we can delay CP2K release by ~1 week (just in case everything goes wrong). The new DBCSR will support HIP and OpenCL + some optimizations.

**Comment by oschuett** [comment]
Don't worry. We shall certainly wait for DBCSR.

**Comment by alazzaro** [comment]
@oschuett DBCSR 2.3 is out and part of CP2K (dashboard is happy too).

**Comment by alazzaro** [comment]
One more thing you can consider for the new CP2k release: a new COSMA version is almost ready to go (v2.6.0 branch). It has a nice G2G layer implemented by @kabicm

**Comment by oschuett** [comment]
Yes, I also saw that a new [SIRIUS release](https://github.com/electronic-structure/SIRIUS/releases/tag/v7.3.2) is coming soon.

However, I'm a little reluctant to 1) hold our release much longer and 2) release shortly after upgrading without much field testing. What do you think @toxa81?

**Comment by kabicm** [comment]
Thanks for the invite! Yes, the new version can use NCCL/RCCL and GPU-aware MPI. It would be nice if you can test cp2k with COSMA v2.6.0 branch, so that we can debug if some tests fail before a new release. Would it be possible?

**Comment by toxa81** [comment]
We tagged a new pre-release to test/fix the lda+u interface. I would prefer to keep the current version of sirius in the release, but switch to 7.3.2 in the develop branch and test it carefully.

**Comment by oschuett** [comment]
>  It would be nice if you can test cp2k with COSMA v2.6.0 branch, so that we can debug if some tests fail before a new release. Would it be possible?

Yes, you can open a draft pull request or run the tests locally via Docker:
```
cd cp2k/tools/docker
docker build --shm-size=1g -f Dockerfile.test_sdbg -t cp2k_test_sdbg ../../
```

**Comment by oschuett** [comment]
Ok then I would suggest that we release CP2K as is.

Afterwards we can then take our time for the upgrades. I can also help with adding GPU-away MPI to the toolchain because I'll need it for DBM too.

**Comment by kabicm** [comment]
This makes sense -- it's better to avoid the stress of a new release in a short time period! :)

So, release CP2K as you planned and then let's work together on updating to the new cosma version.
